README_ENGINE.md

```diff
@@ -37,55 +45,204 @@ The engine source code can be found in the `/engine` folder. The source code is
 * **tools** - Tools such as the gamepad configurator
 * **webview** - WebView stub implementation. Actual implementation moved to [extension-webview](https://github.com/defold/extension-webview)
 
+</details>
+
+## Code
+
+One of our goals is to keep the engine as small and performant as possible.
+By this we mean both development time, and actual runtime for the released game:
+compiled code size, source code size, compile time, runtime.
+
+By thinking this in mind whenever we add new features, we can keep this goal with minimal effort.
+
+### When to add code?
+
+The fastest code is the code that isn't run, or isn't even added!
+
+If we don't need it, don't add it.
+
+Think of _all_ the end users. Will the fix/feature have a clear benefit to most of our users?
+If not, can the feature be added as a separate native extension instead?
+
+### Design
+
+Before adding code to the engine, the code needs a design first.
```

Comment:
I suggest adding this as a "#### Features" section, to emphasize that for new features we really want to see a design (and maybe even a community discussion first). For Fixes the bar is obviously lower.

Reply:
... then we can also say "before adding a feature to the engine, the feature...."